lib/constructs/interface.js

```diff
@@ -26,6 +26,35 @@ Interface.prototype.implements = function (source) {
   this.mixins.push(source);
 };
 
+Interface.prototype.generateNamedPropertiesObject = function () {
+  if (!utils.isGlobal(this.idl)) {
+    return;
+  }
+
+  this.str += `var ${this.name}PropertiesConstructor = function () {
```

Comment:
As far as I can tell there's no constructor for the named properties object. It's just a singleton object. So the proxy should have as its target an empty object (probably an `Object.create(null)` object)